heron/spi/src/java/com/twitter/heron/spi/packing/PackingPlan.java

```diff
@@ -123,11 +123,15 @@ public Integer getInstanceCount() {
    * @return String describing component ram distribution
    */
   public String getComponentRamDistribution() {
+    // Generate a map with the minimal ram size for each component
     Map<String, ByteAmount> ramMap = new HashMap<>();
-    // The implementation assumes instances for the same component require same ram
     for (ContainerPlan containerPlan : this.getContainers()) {
       for (InstancePlan instancePlan : containerPlan.getInstances()) {
-        ramMap.put(instancePlan.getComponentName(), instancePlan.getResource().getRam());
+        ByteAmount newRam = instancePlan.getResource().getRam();
+        ByteAmount currentRam = ramMap.get(instancePlan.getComponentName());
+        if (currentRam == null || currentRam.asBytes() > newRam.asBytes()) {
```

Comment:
why newRam != currentRam

Reply:
It happens to all algorithms in fact, but it may not be a problem because some algorithms requires per component resources specified.

And it is impossible to be handled in RoundRobin. Different container has different set of components. RoundRobin calculates the components with specified Ram sizes first and allocate the memory left to the rest components. As the result, instances in different containers quite often have different memory size. The example in the description is the simplest case.